### PR TITLE
planner: skip MDL when analyzing table

### LINKED ISSUE #47475: analyze table shouldn't not block DDL
## Enhancement
Currently, if the user executes `analyze table`, it blocks the DDL on this table by MDL.
Since `analzye table` is a read-only operation, we can let MDL ignore it.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #47475

Problem Summary:

### What changed and how does it work?
Skip lock if the statement is analyze stmt

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Analyze doesn't block DDL any more 
```

### CODE REVIEW

File: pkg/ddl/tests/metadatalock/mdl_test.go [hunk 1]
```diff
@@ -432,6 +432,47 @@ func TestMDLAutoCommitReadOnly(t *testing.T) {
...
+	go func() {
+		tk.MustExec("begin")
+		tk.MustExec("analyze table t;")
+		tk.MustQuery("select sleep(2);")
+		tk.MustExec("commit")
+		ts1 = time.Now()
+		wg.Done()
+	}()
+
+	go func() {
```
Comment (Benjamin2037) [inline pkg/ddl/tests/metadatalock/mdl_test.go:466]: add partition table-related test case？for example add partition

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by Benjamin2037** [comment]
/cherry-pick https://github.com/pingcap/tidb/tree/release-6.5-20230625-v6.5.3

**Pull request merged**